Add the user guide for upgrading v0.11 to v0.12 (#2531)
apache/iotdb: docs/Download/README.md

**<font color=red>Attention</font>**:

- How to upgrade a minor version (e.g., from v0.11.0 to v0.11.1)?
  * versions which have the same major version are compatible.
  * Just download and unzip the new version. Then modify the configuration files to keep consistent 
  with what you set in the old version.
  * stop the old vesion instance, and start the new one.

- How to upgrade from v.11.x or v0.10.x to v0.12.x?
  * Upgrading from v0.11 or v0.10 to v0.12 is similar as v0.9 to v0.10. We the upgrade tool will rewrite the data files.
  * Stop writing new data.
  * Call `flush` command using sbin/start-cli.sh in original version to close all TsFiles.
  * We recommend to backup the data file (also the wal files and mlog.txt) before upgrading for rolling back.
  * Just download, unzip v0.12.x.zip, and modify conf/iotdb-engine.proeprties to let all the 
  directories point to the folders set in previous version (or the backup folder). 
  You can also modify other settings if you want. 

>   * Stop IoTDB v0.11 or v0.10 instance, and start v0.12.x, then the IoTDB will upgrade data file format automatically. We are able to read and write when data files are upgrading.

Reviewer: > We are able to read and write when data files are upgrading.

It is ok to read and write data when the upgrading process works.

Author: ```suggestion
  * Stop IoTDB v0.11 or v0.10 instance, and start v0.12.x, then the IoTDB will upgrade data file format automatically. It is ok to read and write data when the upgrading process works.
```
